File: crates/target_java_jackson/output/description/RootDiscriminatorWithDescription.java
package com.example;


import com.fasterxml.jackson.annotation.JsonSubTypes;

import com.fasterxml.jackson.annotation.JsonTypeInfo;

/**
 * A description for discriminator
 */

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "foo")
@JsonSubTypes({

    @JsonSubTypes.Type(name = "bar", value = RootDiscriminatorWithDescriptionBar.class),

})
public abstract class RootDiscriminatorWithDescription {
}

File: crates/target_csharp_system_text/output/description/RootDiscriminatorWithDescription.cs
using System;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace JtdCodegenE2E
{
    /// <summary>
    /// A description for discriminator
    /// </summary>
    [JsonConverter(typeof(RootDiscriminatorWithDescriptionJsonConverter))]
    public abstract class RootDiscriminatorWithDescription
    {
    }

    public class RootDiscriminatorWithDescriptionJsonConverter : JsonConverter<RootDiscriminatorWithDescription>
    {
        public override RootDiscriminatorWithDescription Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
        {
            var readerCopy = reader;
            var tagValue = JsonDocument.ParseValue(ref reader).RootElement.GetProperty("foo").GetString();

            switch (tagValue)
            {
                case "bar":
                    return JsonSerializer.Deserialize<RootDiscriminatorWithDescriptionBar>(ref readerCopy, options);
                default:
                    throw new ArgumentException(String.Format("Bad Foo value: {0}", tagValue));
            }
        }

        public override void Write(Utf8JsonWriter writer, RootDiscriminatorWithDescription value, JsonSerializerOptions options)
        {
            JsonSerializer.Serialize(writer, value, value.GetType(), options);
        }
    }
}
